qiskit/algorithms/factorizers/shor.py

```diff
@@ -50,6 +56,10 @@ class Shor:
     See also https://arxiv.org/abs/quant-ph/0205095
     """
 
+    @deprecate_function(
```

Comment:
Right now this will trigger a `DeprecationWarning`, should we use a `PendingDeprecationWarning` first and remove these algorithms in 6 months instead of 3 since this is quite a substantial removal?

Reply:
They get removed after a min of 3 months or two releases right. So they will be around for 6 months roughly if deprecated now. The other algorithms such as VQE etc, I would have expected these to be deprecated now favoring the new primitive based ones that will be introduced shortly so I would have though things consistent with that.

Reply:
For this where there isn't an alternative it's a deprecation towards removal since we're saying they're no longer in scope for terra deprecation up front is fine. For things like vqe though @Cryoris is correct since we're replacing that functionality with a new thing we have to have the new thing in a release prior to emitting a deprecation warning.

For more details you can refer to https://qiskit.org/documentation/deprecation_policy.html